plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,410 @@
+		// This is what we want to be able to do, eventually:
+		//     telemetry.GetMeasurements()[fieldName] = float64Value
+		// (plus handle name conflics)
+		addUniquelyNamedProperty(properties, fieldName, fmt.Sprintf("%g", float64Value))
+	}
+}
+
+func contains(set []string, val string) bool {
+	for _, elem := range set {
+		if elem == val {
+			return true
+		}
+	}
+
+	return false
+}
+
+func addUniquelyNamedProperty(properties map[string]string, name string, value string) {
+	if _, found := properties[name]; !found {
+		properties[name] = value
+	} else {
+		suffix := 1
+		for {
+			_, found = properties[name+"_"+strconv.Itoa(suffix)]
+			if !found {
+				break
+			}
+			suffix++
+		}
+		properties[name+"_"+strconv.Itoa(suffix)] = value
+	}
+}
+
+func toFloat64(value interface{}) (float64, error) {
+	switch v := value.(type) {
+	case int:
+		return float64(v), nil
+	case int8:
+		return float64(v), nil
+	case int16:
+		return float64(v), nil
+	case int32:
+		return float64(v), nil
+	case int64:
+		return float64(v), nil
+	case uint:
+		return float64(v), nil
+	case uint8:
+		return float64(v), nil
+	case uint16:
+		return float64(v), nil
+	case uint32:
+		return float64(v), nil
+	case uint64:
+		return float64(v), nil
+	case float32:
+		return float64(v), nil
+	case float64:
+		return v, nil
+	}
+
+	return 0.0, fmt.Errorf("[%s] cannot be converted to a float64 value", value)
+}
+
+func toInt(value interface{}) (int, error) {
+	if !is32BitChecked {
+		is32BitChecked = true
+		i := int(-1)
+		j := uint(i)
+		is32Bit = (j == math.MaxUint32)
+	}
+
+	switch v := value.(type) {
+	case int:
+		return v, nil
+	case int8:
+		return int(v), nil
+	case int16:
+		return int(v), nil
+	case int32:
+		return int(v), nil
+	case uint8:
+		return int(v), nil
+	case uint16:
+		return int(v), nil
+
+	case uint:
+		if is32Bit {
+			if v > math.MaxInt32 {
+				return 0, fmt.Errorf("Value [%d] out of range of 32-bit integers", v)
+			}
+		} else {
+			if uint64(v) > math.MaxInt64 {
+				return 0, fmt.Errorf("Value [%d] out of range of 64-bit integers", v)
+			}
+		}
+
+		return int(v), nil
+
+	case uint32:
+		if is32Bit {
+			if v > math.MaxInt32 {
+				return 0, fmt.Errorf("Value [%d] out of range of 32-bit integers", v)
+			}
+		}
+
+		return int(v), nil
+
+	case uint64:
+		if is32Bit {
+			if v > math.MaxInt32 {
+				return 0, fmt.Errorf("Value [%d] out of range of 32-bit integers", v)
+			}
+		} else {
+			if v > math.MaxInt64 {
+				return 0, fmt.Errorf("Value [%d] out of range of 64-bit integers", v)
+			}
+		}
+
+		return int(v), nil
+
+	case int64:
+		if is32Bit {
+			if v > math.MaxInt32 || v < math.MinInt32 {
+				return 0, fmt.Errorf("Value [%d] out of range of 32-bit integers", v)
+			}
+		}
+
+		return int(v), nil
+	}
+
+	return 0.0, fmt.Errorf("[%s] cannot be converted to an int value", value)
+}
+
+func logOutputMsg(format string, v ...interface{}) {
+	log.Printf("Output [application_insights] "+format, v...)
```

Comment:
nit: space between '+'

Reply:
This is how standard go code formatting formats this expression